lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -538,6 +538,69 @@ class TypeRefinementContextBuilder : private ASTWalker {
     return MacroWalking::Arguments;
   }
 
+  /// Check whether this declaration is in a source file buried within
+  /// a macro expansion of the
```

Comment:
Looks like your thought wasn't finished here

Reply:
Thank you, finished the thought in https://github.com/swiftlang/swift/pull/77528